kombu/transport/sqlalchemy/__init__.py

```diff
@@ -209,7 +209,12 @@ def _purge(self, queue):
         return count
 
     def _size(self, queue):
-        return self._query_all(queue).count()
+        obj = self._get_or_create(queue)
+        return (
+            self.session.query(self.message_cls)
+            .filter(self.message_cls.queue_id == obj.id)
+            .count()
+        )
```

Comment:
should the count include visible=false messages by the way? looks like it's only used for `queue_declare_ok_t`?

Reply:
Yes please. Invisible msgs should not be counted.